src/Microsoft.AspNetCore.Authentication.Impl/AuthenticationServiceCollectionExtensions.cs

```diff
@@ -46,8 +45,7 @@ public static IServiceCollection AddAuthentication(this IServiceCollection servi
             return services;
         }
 
-        // REVIEW: rename to just AddScheme?
-        public static IServiceCollection AddSchemeHandler<TOptions, THandler>(this IServiceCollection services, string authenticationScheme, Action<TOptions> configureOptions)
+        public static IServiceCollection AddScheme<TOptions, THandler>(this IServiceCollection services, string authenticationScheme, Action<TOptions> configureOptions, bool canHandleRequests)
```

Comment:
Replacing `IAuthenticationSchemeProvider` is definitely not an option (it will be broken if the user decides to register his own instance).

Having to create a custom middleware just for that seems weird. I thought the whole point of this PR was to remove the "authentication middleware" concept (except the new global one, of course)? :open_mouth: